protocols/imap/src/test/java/org/apache/james/imap/processor/ListProcessorTest.java

```diff
@@ -19,130 +19,138 @@
-        mockery.checking(new Expectations() {{
-            oneOf(responder).respond(with(equal(createResponse(false, true, false, false, false, false, '.', "INBOX"))));
-        }});
-        processor.processResult(responder, false, result,MailboxType.OTHER);
+    public void convertNoSelectUnknownChildrenShouldHaveNoSelectFlagOnly() throws Exception {
+        ListResponse actual = processor.convertMetadataToListResponse(false,
+            new SimpleMailboxMetaData(inboxPath, null, '.', MailboxMetaData.Children.CHILDREN_ALLOWED_BUT_UNKNOWN, MailboxMetaData.Selectability.NOSELECT),
+            MailboxType.OTHER);
+        assertThat(actual).isEqualTo(
+            new ListResponse(
+                false,
+                true,
+                false,
+                false,
+                false,
+                false,
+                inboxPath.getFullName('.'),
+                '.')
+        );
     }
 
     @Test
-    public void testUnMarked() throws Exception {
-        setUpResult(MailboxMetaData.Children.CHILDREN_ALLOWED_BUT_UNKNOWN, MailboxMetaData.Selectability.UNMARKED, '.', inboxPath);
-        mockery.checking(new Expectations() {{
-            oneOf(responder).respond(with(equal(createResponse(false, false, false, true, false, false, '.', "INBOX"))));
-        }});
-        processor.processResult(responder, false, result,MailboxType.OTHER);
+    public void convertUnmarkedUnknownChildrenShouldHaveUnmarkedFlagOnly() throws Exception {
+        ListResponse actual = processor.convertMetadataToListResponse(false,
+            new SimpleMailboxMetaData(inboxPath, null, '.', MailboxMetaData.Children.CHILDREN_ALLOWED_BUT_UNKNOWN, MailboxMetaData.Selectability.UNMARKED),
+            MailboxType.OTHER);
+        assertThat(actual).isEqualTo(
+            new ListResponse(
+                false,
+                false,
+                false,
+                true,
+                false,
+                false,
+                inboxPath.getFullName('.'),
+                '.')
+        );
     }
 
     @Test
-    public void testMarked() throws Exception {
-        setUpResult(MailboxMetaData.Children.CHILDREN_ALLOWED_BUT_UNKNOWN, MailboxMetaData.Selectability.MARKED, '.', inboxPath);
-        mockery.checking(new Expectations() {{
-            oneOf(responder).respond(with(equal(createResponse(false, false, true, false, false, false, '.', "INBOX"))));
-        }});
-        processor.processResult(responder, false, result,MailboxType.OTHER);
+    public void convertMarkedUnknownChildrenShouldHaveMarkedFlagOnly() throws Exception {
+        ListResponse actual = processor.convertMetadataToListResponse(false,
+            new SimpleMailboxMetaData(inboxPath, null, '.', MailboxMetaData.Children.CHILDREN_ALLOWED_BUT_UNKNOWN, MailboxMetaData.Selectability.MARKED),
+            MailboxType.OTHER);
+        assertThat(actual).isEqualTo(
+            new ListResponse(
+                false,
```

Comment:
One whould say there is a lot of magic contants.

Reply:
what would you suggest ? I mean, it's the tested output.

Reply:
:-) I would suggest to  have a simpler test output.